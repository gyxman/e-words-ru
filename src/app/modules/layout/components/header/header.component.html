<div class="info">
    <button
        automation-id="toggle"
        mat-icon-button
        color="primary"
        aria-label="Открыть меню"
    >
        <mat-icon
            svgIcon="toggle"
            aria-hidden="false"
            aria-label="Иконка открытия меню"
        ></mat-icon>
    </button>
    <a class="logo" automation-id="logo" [routerLink]="['/']">Тренажер слов</a>
</div>
<div class="actions">
    <button
        automation-id="exit"
        mat-icon-button
        color="primary"
        aria-label="Выйти из приложения"
    >
        <mat-icon
            svgIcon="logout"
            aria-hidden="false"
            aria-label="Иконка выхода из приложения"
        ></mat-icon>
    </button>
</div>
