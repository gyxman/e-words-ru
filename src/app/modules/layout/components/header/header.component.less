@import '../../../../../styles/index';

:host {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 12px 16px;
    background-color: @color-gray-extra-light2;
}

.info {
    display: flex;
    align-items: center;
}

.logo {
    .font-h4();
    margin-left: 16px;
    text-decoration: none;
    color: unset;
}
