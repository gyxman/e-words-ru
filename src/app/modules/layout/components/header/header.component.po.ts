import {PageObject} from '../../../../testing/page-object';
import {DebugElement} from '@angular/core';

export class HeaderComponentPo<T> extends PageObject<T> {
    get openMenu(): DebugElement {
        return this.getByAutomationId('open-menu');
    }

    get exit(): DebugElement {
        return this.getByAutomationId('exit');
    }
}
