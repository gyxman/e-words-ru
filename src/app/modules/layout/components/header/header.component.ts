import {Component} from '@angular/core';
import {MatIconRegistry} from '@angular/material/icon';
import {DomSanitizer} from '@angular/platform-browser';

@Component({
    selector: 'app-header',
    templateUrl: './header.component.html',
    styleUrls: ['./header.component.less'],
})
export class HeaderComponent {
    constructor(iconRegistry: MatIconRegistry, sanitizer: DomSanitizer) {
        iconRegistry.addSvgIcon(
            'logout',
            sanitizer.bypassSecurityTrustResourceUrl('assets/icons/icon-logout.svg'),
        );
        iconRegistry.addSvgIcon(
            'toggle',
            sanitizer.bypassSecurityTrustResourceUrl('assets/icons/icon-toggle.svg'),
        );
    }
}
