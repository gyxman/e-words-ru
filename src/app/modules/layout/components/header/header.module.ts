import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {HeaderComponent} from './header.component';
import {MatButtonModule} from '@angular/material/button';
import {MatIconModule} from '@angular/material/icon';
import {RouterModule} from '@angular/router';

@NgModule({
    imports: [CommonModule, MatButtonModule, MatIconModule, RouterModule],
    declarations: [HeaderComponent],
    exports: [HeaderComponent],
})
export class HeaderModule {}
