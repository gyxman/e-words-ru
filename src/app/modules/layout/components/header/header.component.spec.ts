import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {HeaderComponent} from './header.component';
import {MockDirective, MockModule} from 'ng-mocks';
import {MatButtonModule} from '@angular/material/button';
import {MatIconModule} from '@angular/material/icon';
import {HeaderComponentPo} from './header.component.po';
import {AuthFacadeService} from '../../../auth/services/auth-facade.service';
import {deepEqual, instance, mock, verify} from 'ts-mockito';
import {Router, RouterLink} from '@angular/router';
import {AuthRouteEnum} from '../../../auth/enums/auth-route.enum';

describe('HeaderComponent - компонент верхней части авторизованной зоны', () => {
    let component: HeaderComponent;
    let fixture: ComponentFixture<HeaderComponent>;
    let pageObject: HeaderComponentPo<HeaderComponent>;
    let authFacadeServiceMock: AuthFacadeService;
    let routerMock: Router;

    beforeEach(() => {
        authFacadeServiceMock = mock(AuthFacadeService);
        routerMock = mock(Router);
    });

    beforeEach(async(() => {
        TestBed.configureTestingModule({
            declarations: [HeaderComponent, MockDirective(RouterLink)],
            imports: [MockModule(MatButtonModule), MockModule(MatIconModule)],
            providers: [
                {
                    provide: AuthFacadeService,
                    useFactory: () => instance(authFacadeServiceMock),
                },
                {
                    provide: Router,
                    useFactory: () => instance(routerMock),
                },
            ],
        }).compileComponents();
    }));

    beforeEach(() => {
        fixture = TestBed.createComponent(HeaderComponent);
        component = fixture.componentInstance;
        pageObject = new HeaderComponentPo(fixture);
    });

    it('Если компонент загрузился, то все элементы отрисовались', () => {
        // act
        fixture.detectChanges();

        // assert
        expect(fixture.nativeElement).toMatchSnapshot();
    });

    it('Если пользователь нажимает на кнопку выхода из приложения, вызывается метод выхода', () => {
        // arrange
        fixture.detectChanges();

        // act
        pageObject.click(pageObject.exit);

        // assert
        verify(authFacadeServiceMock.signOut()).once();
    });

    it('Если пользователь нажимает на кнопку выхода из приложения, перебрасываем пользователя на страницу логина', () => {
        // arrange
        fixture.detectChanges();

        // act
        pageObject.click(pageObject.exit);

        // assert
        verify(routerMock.navigate(deepEqual([AuthRouteEnum.Login]))).once();
    });
});
