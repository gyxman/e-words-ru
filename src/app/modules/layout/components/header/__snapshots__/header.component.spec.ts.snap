// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent - компонент верхней части авторизованной зоны Если компонент загрузился, то все элементы отрисовались 1`] = `
<div
  id="root0"
>
  <div
    class="info"
  >
    <button
      aria-label="Открыть меню"
      automation-id="toggle"
      color="primary"
      mat-icon-button=""
    >
      <mat-icon
        aria-hidden="false"
        aria-label="Иконка открытия меню"
        svgicon="toggle"
      />
    </button>
    <a
      automation-id="logo"
      class="logo"
      routerlink="/"
    >
      Тренажер слов
    </a>
  </div>
  <div
    class="actions"
  >
    <button
      aria-label="Выйти из приложения"
      automation-id="exit"
      color="primary"
      mat-icon-button=""
    >
      <mat-icon
        aria-hidden="false"
        aria-label="Иконка выхода из приложения"
        svgicon="logout"
      />
    </button>
  </div>
</div>
`;
